specification/v1.4.0-rc/OSDM-offline-model-v1.4.0-rc.json

```diff
@@ -1,2246 +1,2362 @@
+								"description": "minutes of the day in the time zone of travel",
+								"type": "integer",
+								"format": "int32"
+							},
+							"until": {
+								"description": "minutes of the day in the time zone of travel",
+								"type": "integer",
+								"format": "int32"
+							},
+							"scope": {
+								"description": "START_OF_TRAVEL,COMPLETE_RANGE",
+								"type": "string"
+							}
+						},
+						"required": [
+							"from",
+							"until",
+							"scope"
+						],
+						"additionalProperties": false
+					}
+				},
+				"numberOfTravelDays": {
+					"description": "number of allowed travel days (e.g. 3 travel days within 2 weeks)",
+					"type": "integer",
+					"format": "int32"
+				},
+				"returnConstraint": {
+					"description": "a return trip with the same carrier must be sold in combination",
+					"type": "object",
+					"properties": {
+						"latestReturn": {
+							"description": "number of days after departure or start of validity of the last return",
+							"type": "integer",
+							"format": "int32"
+						},
+						"earliestReturn": {
+							"description": "number of days after departure or start of validity of the earliest return",
+							"type": "integer",
+							"format": "int32"
+						},
+						"excludedWeekdays": {
+							"description": "weekdays (ISO day of the week, 1 = Monday) between travel and return where travel is not allowed",
+							"type": "array",
+							"items": {
+								"type": "integer",
+								"format": "int32"
+							}
+						}
+					},
+					"required": [
+						"latestReturn",
+						"earliestReturn"
+					],
+					"additionalProperties": false
+				},
+				"trainValidity": {
+					"description": "validity depending on boarding / leaving of a train. The ticket is also valid from departure until the destination station in the train ",
+					"type": "object",
+					"properties": {
+						"carrierConstraintRef": {
+							"type": "string"
+						},
+						"serviceConstraintRef": {
+							"type": "string"
+						},
+						"scope": {
+							"type": "string",
+							"enum": [
+								"BOARDING",
+								"ARRIVAL"
+							],
+							"default": "BOARDING"
+						}
+					},
+					"required": [
+						"carrierConstraintRef",
+						"scope"
+					],
+					"additionalProperties": false
+				},
+				"validityType": {
+					"description": "type of usage, eigther as single trip or multiple trips. Unrestricted fares might be combined into single trips when appropriate",
+					"type": "string",
+					"default": "SINGLE_TRIP",
+					"x-extensible-enum": [
+						"SINGLE_TRIP",
+						"MULTIPLE_TRIPS",
+						"UNRESTRICTED"
+					]
+				},
+				"tripAllocationConstraint": {
+					"$ref": "#/definitions/TripAllocationConstraintDef"
+				},
+				"tripInterruptionConstraint": {
+					"$ref": "#/definitions/TripInterruptionConstraintDef"
+				}
+			},
+			"required": [
+				"validityRange"
+			],
+			"additionalProperties": false
+		},
+		"TripInterruptionProcess": {
```

Comment:
Is there a definition somewhere of what constitutes a trip interruption?

Reply:
Not sure about UIC-endorsed regularions but in Czechia, Ministry of Transport and Ministry of Finance both issue generally required conditions of carriage, and these set the conditions for interruption, e.g. each opet ticket can be interrupted after travelling for 100 or more kilometers. And rest of journey must be completed within ticket time validity which is 0.00 on travel date till 6.00 on the following day (and the validity is also regulated).

Reply:
Local goverments regulate sometimes the domestic tickets and sometimes only the domestic full fare tickets, international tariffs are not covered by these regulations, so the fare providers define them on their own.

Reply:
Of course, regional networks are one different world. Internation fares usually have specific rules as well (train-bound fixed fares, flexible fares, ...) I guest we should not try to express everything in OSDM like we did in refundability of fares - yes, no, with conditions (but details not expressed in the API response).